content/kubermatic/main/tutorials-howtos/monitoring-logging-alerting/master-seed/customization/thanos.md

```diff
@@ -0,0 +1,302 @@
++++
+linkTitle = "Thanos"
+title = "Thanos integration with prometheus"
+date = 2023-07-20T09:44:15+05:30
+weight = 20
++++
+
+This page explains how we can integrated [Thanos](https://thanos.io/) long term storage of metrics with KKP seed prometheus
+
+## Pre-requsites
+1. Helm is installed.
+1. KKP v2.22.4+ is installed in the cluster.
+1. KKP prometheus chart has been deployed in each seed where you want to store long term metrics
+
+## Integration steps
+Below page outlines
+1. Install thanos components in your kubernetes cluster via Helm chart
+1. Customize KKP prometheus chart to augment prometheus pod with thanos sidecar
+2. Customize KKP prometheus chart values to monitor and get alerts for Thanos components
+
+## Install thanos chart
+
+You can install thanos helm chart from bitnami chart repository 
+```shell
+HELM_EXPERIMENTAL_OCI=1 helm upgrade --install thanos \
+  --namespace monitoring --create-namespace\
+   --version 12.8.6 \
+  -f thanos-values.yaml \
+  oci://registry-1.docker.io/bitnamicharts/thanos 
+```
+
+### Basic Thanos Customization file 
+You can configure Thanos to store the metrics in any s3 compatible storage as well as other popular cloud storate solution.
+
+Below yaml snippet uses Azure Blob storage configuration. You can refer to all [supported object storage configurations](https://thanos.io/tip/thanos/storage.md/#supported-clients).
+
+```yaml
+# thanos-values.yaml
+# Refer https://artifacthub.io/packages/helm/bitnami/thanos for more configuration options
+objstoreConfig: |-
+  type: AZURE
+  config:
+    storage_account: "<AZ_STORAGE_ACCT>"
+    storage_account_key: "<AZ_STORAGE_ACCT_KEY>"
+    container: "<AZ blob container name>"
+    max_retries: 0
+compactor:
+  enabled: true
+  persistence:
+    enabled: true
+    # size: 80Gi
+  retentionResolutionRaw: 2d
+  retentionResolution5m: 2w
+  retentionResolution1h: 60d
+metrics:
+  enabled: true
+storegateway:
+  enabled: true
+```
+
+
+## Augment prometheus to use Thanos sidecar
+
+In order to receive metrics from prometheus into Thanos, Thanos provides two mechanisms.
+1. [Thanos Sidecar](https://thanos.io/tip/components/sidecar.md/)
+1. [Thanos Receiver](https://thanos.io/tip/components/receive.md/)
+
+Thanos sidecar is a much simpler and less resource heavy approach. You can understand more about Thanos components [here](https://thanos.io/tip/thanos/quick-tutorial.md/#components). Due to simplicity, we have outlined how to integrate the Thanos sidecar in existing prometheus chart configuration.
```

Comment:
```suggestion
Thanos sidecar is a much simpler and less resource heavy approach. You can laern more about Thanos components [here](https://thanos.io/tip/thanos/quick-tutorial.md/#components). Due to simplicity, we have outlined how to integrate the Thanos sidecar in the existing Prometheus chart configuration.
```

Reply:
Your suggestion had spelling mistake... so... I think I will fix it via manual commit.

Reply:
Check latest push.